From 6287e6f0fdd36be4bbde6f539df6ea85eb2476c2 Mon Sep 17 00:00:00 2001
From: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
Date: Tue, 26 Jul 2022 18:29:35 +0300
Subject: [PATCH] tools/power turbostat: do not decode ACC for ICX and SPR
Git-commit: 6287e6f0fdd36be4bbde6f539df6ea85eb2476c2
Patch-mainline: v6.0-rc1
References: bsn#355
Modified-by-SEL: No


The ACC (automatic C-state conversion) feature was available on Sky Lake and
Cascade Lake Xeons (SKX and CLX), but it is not available on Ice Lake and
Sapphire Rapids Xeons (ICX and SPR). Therefore, stop decoding it for ICX and
SPR.

Signed-off-by: Artem Bityutskiy <artem.bityutskiy@linux.intel.com>
Signed-off-by: Len Brown <len.brown@intel.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 tools/power/x86/turbostat/turbostat.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index 05e993674015..4c23e1fb13e9 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -4822,8 +4822,6 @@ void automatic_cstate_conversion_probe(unsigned int family, unsigned int model)
 	switch (model) {
 	case INTEL_FAM6_BROADWELL_X:
 	case INTEL_FAM6_SKYLAKE_X:
-	case INTEL_FAM6_ICELAKE_X:
-	case INTEL_FAM6_SAPPHIRERAPIDS_X:
 		has_automatic_cstate_conversion = 1;
 	}
 }
-- 
2.34.1

